controller/volume_controller.go

```diff
@@ -528,7 +528,7 @@ func (c *VolumeController) syncVolume(key string) (err error) {
 	}
 
 	if err := c.ReconcileBackupVolumeState(volume); err != nil {
-		return nil
+		return err
```

Comment:
Actually, I am thinking if the controller can continue the reconciliation after encountering this issue: This function mainly updates the backup-related status fields for volume CRs. Would you leave them with the previous values and continue updating other fields if there is a backup volume error?

Reply:
WDYT? @PhanLe1010

Reply:
Sorry for the delay. With the PR https://github.com/longhorn/longhorn-manager/pull/3893/files already trying to clean up duplicated backupvolume CR, I think it is best to simply return err here and retry later. The douplicated backup volume CR would be cleaned up and the normal volume sync follow would happen. WDYT @derekbit @shuo-wu ?

Reply:
Then, @PhanLe1010's impl LGTM.